lib/http-store.js

```diff
@@ -0,0 +1,134 @@
+'use strict'
+const request = require('request');
+const util = require('util');
+const stormer = require('stormer');
+const assert = require('assert');
+const qs = require('qs');
+
+const Store = stormer.Store;
+const NotFoundError = stormer.errors.NotFoundError;
+
+/**
+ * @class HttpStore
+ * @extends Store
+ *
+ */
+class HttpStore extends Store {
+
+  /**
+   * Generates an HttpStore instance
+   *
+   * @constructor
+   * @memberof HttpStore
+   * @param {Object} requestClient - The request client to use
+   * @param {Object} options
+   * @return {HttpStore}
+   *
+   */
+  constructor(requestClient = request, options) {
+    assert(request, 'request is required');
+    assert(options, 'options are required');
+    super();
+    this._request = requestClient;
+    this._timeout = options.timeout || 1000;
+    this._baseUrl = options.baseUrl;
+  }
+
+  /**
+   * Wraps an HTTP request within a Promise
+   *
+   * @method _requestPromise
+   * @memberof HttpStore
+   * @param {Object} options
+   * @return {Promise}
+   *
+   */
+  _requestPromise(options) {
+    return new Promise((resolve, reject) => this._request(options, function(err, res) {
+      if (err) {
+        return reject(err);
+      }
+      return resolve(res.body);
+    }));
+  }
+
+  /**
+   * Retrieves an entry from the storage
+   *
+   * @method _get
+   * @memberof HttpStore
+   * @param {Object} model - The model
+   * @param {String} pk - The object's primary key
+   * @return {Promise} - A Promise
+   *
+   */
+  _get(model, pk) {
+    return this._requestPromise({
+      method: 'GET',
+      url: util.format('%s/%s/%s', this._baseUrl, model.name, pk)
+    }).catch(() => {
```

Comment:
Need to evaluate if response is 404 to return `NotFoundError`, if err is related to timeout or 500 it should not be not found error here